vm/vm.go

```diff
@@ -212,13 +212,21 @@ func (vm *VM) Initialize(
 		return fmt.Errorf("failed to initialize p2p: %w", err)
 	}
 
+	blockDBRegistry := prometheus.NewRegistry()
+	if err := vm.snowCtx.Metrics.Register("blockdb", blockDBRegistry); err != nil {
+		return fmt.Errorf("failed to register blockdb metrics: %w", err)
+	}
 	pebbleConfig := pebble.NewDefaultConfig()
-	vm.vmDB, err = storage.New(pebbleConfig, vm.snowCtx.ChainDataDir, blockDB, vm.snowCtx.Metrics)
+	vm.vmDB, err = storage.New(pebbleConfig, vm.snowCtx.ChainDataDir, blockDB, blockDBRegistry)
 	if err != nil {
 		return err
 	}
 
-	vm.rawStateDB, err = storage.New(pebbleConfig, vm.snowCtx.ChainDataDir, stateDB, vm.snowCtx.Metrics)
+	rawStateDBRegistry := prometheus.NewRegistry()
+	if err := vm.snowCtx.Metrics.Register("rawstatedb", blockDBRegistry); err != nil {
+		return fmt.Errorf("failed to register blockdb metrics: %w", err)
```

Comment:
1. Looks like a copy-paste error here and seems like we should be using `rawStateDBRegistry` + in the log.
2. I really don't know much about prometheus - I'm not sure when we're supposed to be sharing a metrics registry vs creating independent ones.

Reply:
I think that we should differentiate db metrics for distinct components ie. these are two different db instances with very different workloads, so conflating them will produce much less useful data.